src/SPC/command/CraftCommand.php

```diff
@@ -0,0 +1,164 @@
+                    $craft['download-options']['ignore-cache-sources'] = 'php-src';
+                } elseif ($craft['download-options']['ignore-cache-sources'] !== null) {
+                    $craft['download-options']['ignore-cache-sources'] .= ',php-src';
+                }
+            }
+            $download_options = $craft['download-options'];
+            foreach ($download_options as $option => $val) {
+                if (is_bool($val) && $val || is_null($val)) {
+                    $args[] = "--{$option}";
+                } elseif (is_string($val)) {
+                    $args[] = "--{$option}={$val}";
+                } elseif (is_array($val)) {
+                    foreach ($val as $v) {
+                        if (is_string($v)) {
+                            $args[] = "--{$option}={$v}";
+                        }
+                    }
+                }
+            }
+            $retcode = $this->runCommand('download', ...$args);
+            if ($retcode !== 0) {
+                $this->output->writeln('<error>craft download failed</error>');
+                $this->log('craft download failed with code: ' . $retcode, true);
+                return static::FAILURE;
+            }
+        }
+
+        // craft build
+        if ($craft['craft-options']['build']) {
+            $args = [$extensions, "--with-libs={$libs}", ...array_map(fn ($x) => "--build-{$x}", $craft['sapi'])];
+            $retcode = $this->runCommand('build', ...$args);
+            if ($retcode !== 0) {
+                $this->output->writeln('<error>craft build failed</error>');
+                $this->log('craft build failed with code: ' . $retcode, true);
+                return static::FAILURE;
+            }
+        }
+
+        return 0;
+    }
+
+    public function processLogCallback($type, $buffer): void
+    {
+        if ($type === Process::ERR) {
+            fwrite(STDERR, $buffer);
+        } else {
+            fwrite(STDOUT, $buffer);
+            $this->log($buffer);
+        }
+    }
+
+    private function log(string $log, bool $master_log = false): void
+    {
+        if ($master_log) {
+            $log = "\n[static-php-cli]> " . $log . "\n\n";
+        } else {
+            $log = preg_replace('/\x1b\[[0-9;]*m/', '', $log);
+        }
+        file_put_contents('craft.log', $log, FILE_APPEND);
+    }
+
+    private function runCommand(string $cmd, ...$args): int
+    {
+        global $argv;
+        if ($this->getOption('debug')) {
+            array_unshift($args, '--debug');
+        }
+        $prefix = PHP_SAPI === 'cli' ? [PHP_BINARY, $argv[0]] : [$argv[0]];
+
+        $process = new Process([...$prefix, $cmd, '--no-motd', ...$args], timeout: null);
+        $this->log("Running: {$process->getCommandLine()}", true);
+
+        if (PHP_OS_FAMILY === 'Windows') {
+            sapi_windows_set_ctrl_handler(function () use ($process) {
+                $process->signal(SIGINT);
+            });
```

Comment:
SIGINT is defined in ext-pcntl
$process->signal sends a POSIX signal. I don't think Windows has POSIX signals, or signals at all?

Reply:
Good catch. Let me look for a solution for Windows.

Reply:
Windows does not support signals, but `$process->signal()` is supported on Windows. When it detects that the directory separator is `\`, it calls taskkill.